src/components/ComparisonMap/ComparisonMap.vue

```diff
@@ -0,0 +1,111 @@
+<template>
+  <div class="comparison-map">
+    <div id="comparison-map-container" class="comparison-map__map-container">
+      <ComparisonDetailMap
+        v-if="reservoirs.length"
+        :reservoirs="reservoirs"
+        :satellite-image-url="oldSatelliteImageUrl"
+        class="comparison-map__detail-map"
+        @setMap="onSetOldMap"
+      />
+
+      <ComparisonDetailMap
+        v-if="reservoirs.length"
+        :reservoirs="reservoirs"
+        :satellite-image-url="satelliteImageUrl"
+        class="comparison-map__detail-map"
+        @setMap="onSetCurrentMap"
+      />
+    </div>
+
+    <div class="comparison-map__dates">
+      <span>{{ parseDate(oldDate) }}</span>
+      <span>{{ parseDate(date) }}</span>
+    </div>
+  </div>
+</template>
+
+<script>
+  import debounce from 'lodash.debounce'
+
+  const DEBOUNCE_TIME = 1000
+  let oldMap, currentMap
+
+  export default {
+    props: {
+      reservoirs: {
+        type: Array,
+        required: true,
+      },
+    },
+
+    data () {
+      return {
+        showComparison: false,
+        oldSatelliteImageUrl: '',
+        satelliteImageUrl: '',
+      }
+    },
+
+    computed: {
+      date () {
+        return new Date()
+      },
+      oldDate () {
+        const date = new Date()
+        date.setFullYear(date.getFullYear() - 1)
+        return date
+      },
+    },
+
+    async mounted () {
+      this.loadSatelliteImages()
+      await this.initializeMap()
+    },
+
+    methods: {
+      parseDate (date) {
+        const options = { year: 'numeric', month: 'long', day: 'numeric' }
+        return date.toLocaleDateString('en-EN', options)
+      },
+      onSetCurrentMap (map) {
+        currentMap = map
+      },
+      onSetOldMap (map) {
+        oldMap = map
+      },
+      async initializeMap () {
+        const MapboxCompare = (await import('mapbox-gl-compare')).default
```

Comment:
It was the only way we found to avoid the different issues described here https://github.com/mapbox/mapbox-gl-compare/issues/1
A plugin, which would be the nicest way Vue-wise, also didn't work because `Vue.use(MapboxCompare)` already tries to call the constructor.
I have added a comment referencing the issue, is that ok?